src/equations/laplace_diffusion_2d.jl

```diff
@@ -0,0 +1,46 @@
+@doc raw"""
+    LaplaceDiffusion2D{T} <: AbstractParabolicEquations{2, 1}
+
+`LaplaceDiffusion2D` represents a scalar diffusion term ``\nabla \cdot (\kappa\nabla u))``
+applied to each solution component.
+"""
+struct LaplaceDiffusion2D{T} <: AbstractLaplaceDiffusionEquations{2, 1}
+  diffusivity::T
+end
+
+# no orientation specified since the flux is vector-valued
+function flux(u, grad_u, equations::LaplaceDiffusion2D)
```

Comment:
Ahh...I think I see what you mean by orientation. The implementation in this PR returns all components/orientations of the viscous flux at each point. In contrast, the implementation in #378 just returns a single `x` or `y` component of the viscous flux at each point - is that correct?

For example, if we have a diffusive matrix `div(A * grad(u))`, I think this would be implemented in this PR's as
```julia
function flux(u, gradients, equations::LaplaceDiffusion2D)
  dudx, dudy = grad_u
  return (equations.A[1,1] * gradients[1] + equations.A[1,2] * gradients[2], 
          equations.A[2,1] * gradients[1] + equations.A[2,2] * gradients[2])
end
```
and in #378's as
```julia
@inline function calcflux(u, gradients, orientation, equation::LinearAdvectionDiffusionEquation2D)
  return -(equation.A[orientation, 1]*gradients[1] + equation.A[orientation, 2]*gradients[2])
end
```

This PR's approach was more natural to me, but I can see that the approach in #378 allows for more flexibility (for example, computing only a single `x` or `y` component at a time, which could get reduce the preallocated memory in cache slightly).

@ranocha what do you think?

Reply:
Maybe add it to the list in #1147 and see how it fits to the `TreeMesh` version?